src/backend/distributed/executor/multi_executor.c

```diff
@@ -131,16 +152,16 @@ CitusExecutorRun(QueryDesc *queryDesc,
 				 ScanDirection direction, uint64 count, bool execute_once)
 {
 	DestReceiver *dest = queryDesc->dest;
-	int originalLevel = FunctionCallLevel;
+	int originalLevel = SPILevel;
 
 	if (dest->mydest == DestSPI)
 	{
 		/*
-		 * If the query runs via SPI, we assume we're in a function call
+		 * If the query runs via SPI, we assume we're in a function call.
 		 * and we should treat statements as part of a bigger transaction.
 		 * We reset this counter to 0 in the abort handler.
 		 */
-		FunctionCallLevel++;
+		SPILevel++;
```

Comment:
is SPILevel still necessary if we have ExecutorLevel? I assume ExecutorLevel gets raised for SPI as well?

Reply:
Removed SPILevel.